src/custom-elements/src/components/text-input/text-input.tsx

```diff
@@ -0,0 +1,165 @@
+
+@Component({
+  tag: 'chi-text-input',
+  styleUrl: 'text-input.scss',
+  scoped: true
+})
+
+export class TextInput {
+  @Element() el: HTMLChiTextInputElement;
+  /**
+   * To define type of Text input
+   */
+  @Prop() type: TextInputTypes = 'text';
+  /**
+   * To define state color of Text input
+   */
+  @Prop({ reflect: true }) state: ChiStates;
+  /**
+   * To add a left positioned icon
+   */
+  @Prop({ reflect: true }) iconLeft: string;
+  /**
+   * To add a right positioned icon
+   */
+  @Prop({ reflect: true }) iconRight: string;
+  /**
+   * To define color of left icon
+   */
+  @Prop({ reflect: true }) iconLeftColor: IconColors;
+  /**
+   * To define color of right icon
+   */
+  @Prop({ reflect: true }) iconRightColor: IconColors;
+  /**
+   * To define size of Text input
+   */
+  @Prop({ reflect: true }) size ?: TextInputSizes = 'md';
+  /**
+   * To define placeholder of Text input
+   */
+  @Prop({ reflect: true }) placeholder: string;
+  /**
+   * To define value of Text input
+   */
+  @Prop({ mutable: true, reflect: true }) value = '';
+  /**
+   * To define name of Text input
+   */
+  @Prop({ reflect: true }) name: string;
+  /**
+   * To disable Text input
+   */
+  @Prop({ reflect: true }) disabled = false;
+  /**
+   * To disable Value attribute mutation
+   */
+  @Prop({ reflect: true }) preventValueMutation = false;
+
+  @State() status: string;
+
+  @Watch('state')
+  stateValidation(newValue: ChiStates) {
+    const validValues = CHI_STATES.join(', ');
+
+    if (newValue && !CHI_STATES.includes(newValue)) {
+      throw new Error(`${newValue} is not a valid state for input. If provided, valid values are: ${validValues}. `);
+    }
+  }
+
+  @Watch('size')
+  sizeValidation(newValue: TextInputSizes) {
+    const validValues = TEXT_INPUT_SIZES.join(', ');
+
+    if (newValue && !TEXT_INPUT_SIZES.includes(newValue)) {
+      throw new Error(`${newValue} is not a valid size for input. If provided, valid values are: ${validValues}. `);
+    }
+  }
+
+  colorValidation(newValue: IconColors) {
+    const validValues = ICON_COLORS.join(', ');
+
+    if (newValue && !ICON_COLORS.includes(newValue)) {
+      throw new Error(`${newValue} is not a valid color for icon. If provided, valid values are: ${validValues}. `);
+    }
+  }
+
+  @Watch('iconLeftColor')
+  iconLeftColorValidation(newValue: IconColors) {
+    this.colorValidation(newValue);
+  }
+
+  @Watch('iconRightColor')
+  iconRightColorValidation(newValue: IconColors) {
+    this.colorValidation(newValue);
+  }
+
+  @Watch('type')
+  typeValidation(newValue: TextInputTypes) {
+    const validValues = TEXT_INPUT_TYPES.join(', ');
+
+    if (newValue && !TEXT_INPUT_TYPES.includes(newValue)) {
+      throw new Error(`${newValue} is not a valid type for input. If provided, valid values are: ${validValues}. `);
+    }
+  }
+
+  /**
+   * To define -hover, -focus statuses
+   */
+  @Method()
```

Comment:
I think the previous approach might be better taking into account the following statement from _stenciljs_ official website: 

> Developers should try to rely on publicly exposed methods as little as possible, and instead default to using properties and events as much as possible. As an app scales, we've found it's easier to manage and pass data through @Prop rather than public methods.

Reply:
@SergioQCL , @janibolkvadze This is intended to be private and will be only used in the testing scenarios. Now, the underscore in the props will be kept undocumented also, Sergio's suggestion will work now. 
In this project, all public stencil methods, props and actions are automatically documented, and added to Chi Docs. Jani hacked the documentation process to don't document props, methods and actions which name is preceded by an underscore. Underscore means private, for Chi developers only, in this case.

Reply:
Yeah, it makes sense to switch it back to props since now we can hide it from the documentation.
Thanks for your help guys!